tests/integration/src/test/java/org/apache/pulsar/tests/integration/cluster/Cluster3Bookie2Broker.java

```diff
@@ -0,0 +1,317 @@
+package org.apache.pulsar.tests.integration.cluster;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.apache.pulsar.tests.PulsarClusterUtils.waitSocketAvaialble;
+
+public class Cluster3Bookie2Broker {
```

Comment:
It would be nice to use the compose stuff in test containers, so that we can also start test clusters manually to try stuff out. I can't see a way to get a handle to individual containers though.

Reply:
compose is an option , but cluster objects are better, all container objects can be accessed by java methods.